src/ui/ui_main.c

```diff
@@ -95,6 +95,18 @@ void UI_MouseEvent(int dx, int dy);
 void UI_Refresh(int realtime);
 qboolean _UI_IsFullscreen(void);
 
+static uiMenuCommand_t UI_AdjustedMenuCommand(const uiMenuCommand_t menutype)
+{
+	// UIMENU_WM_AUTOUPDATE is the only menu we need to adjust,
+	// as it's used in engine. Older versions have it set to 16.
+	if (uiInfo.etLegacyClient > 282010335 || menutype != 16)
+	{
+		return menutype;
+	}
+
+	return UIMENU_WM_AUTOUPDATE;
+}
+
```

Comment:
To further elaborate on this: the **only** menu command that is ever called directly from the engine, that was affected by the re-ordering was `UIMENU_WM_AUTOUPDATE`. There are other calls which set some of the enum values, but they are all below value `12`, which has always had the correct enum order. All other menu commands are sent from the mod code, so they will be correct even though the order has now been changed.

None of these menus are also ever visible during demo playback, therefore we don't need to worry about backwards compatibility with old demos.